source/index.ts

```diff
@@ -553,7 +564,14 @@ export const assert: Assert = {
 	boolean: (value: unknown): asserts value is boolean => assertType(is.boolean(value), 'boolean', value),
 	symbol: (value: unknown): asserts value is symbol => assertType(is.symbol(value), 'symbol', value),
 	numericString: (value: unknown): asserts value is string => assertType(is.numericString(value), AssertionTypeDescription.numericString, value),
-	array: <T = unknown>(value: unknown): asserts value is T[] => assertType(is.array(value), 'Array', value),
+	array: <T = unknown>(value: unknown, assertion?: (element: unknown) => asserts element is T): asserts value is T[] => {
+		const assert: (condition: boolean, description: string, value: unknown) => asserts condition = assertType;
```

Comment:
I did this for prevent the `Assertions require every name in the call target to be declared with an explicit type annotation.ts(2775)` error.

If there is something I am missing here (which I certainly think I am) let me know!